apis/duck/v1alpha1/condition_set.go

```diff
@@ -126,14 +118,23 @@ var _ ConditionManager = (*conditionsImpl)(nil)
 // +k8s:deepcopy-gen=false
 type conditionsImpl struct {
 	ConditionSet
-	accessor ConditionsAccessor
+	status interface{}
 }
 
-// Manage creates a ConditionManager from an object that implements
-// ConditionsAccessopr using the original ConditionSet as a reference.
-func (r ConditionSet) Manage(accessor ConditionsAccessor) ConditionManager {
+// Manage creates a ConditionManager from a status object using the original
+// ConditionSet as a reference. Status must be or point to a struct.
+func (r ConditionSet) Manage(status interface{}) ConditionManager {
```

Comment:
(Surfacing offline discussion)

I think I'd rather have the `conditionsImpl` still operate in terms of a `ConditionsAccessor`, but also supporting the synthesis of a `ConditionsAccessor` via reflection.

That means `Manage` could start taking an `interface{}`, and checking the type via:
```go
  ca, ok := foo.(ConditionsAccessor)
  if !ok {
    // Synthesize a ConditionsAccessor from foo into ca via reflection.
  }
```

Reply:
I think that this separation makes cleanly unit testing the two pieces much easier, and creates a nice functional contract.